Dockerfile

```diff
@@ -1,53 +1,76 @@
-    HOME=/tmp \
+ARG SONAR_SCANNER_VERSION=6.1.0.4477
+ENV HOME=/tmp \
+    XDG_CONFIG_HOME=/tmp \
+    SONAR_SCANNER_HOME=${SONAR_SCANNER_HOME} \
+    SCANNER_BINARIES=https://binaries.sonarsource.com/Distribution/sonar-scanner-cli \
+    SCANNER_ZIP=sonar-scanner.zip
+ENV SCANNER_ZIP_URL="${SCANNER_BINARIES}/sonar-scanner-cli-${SONAR_SCANNER_VERSION}-linux-x64.zip"
+
+WORKDIR /opt
+
+# Security updates
+RUN dnf upgrade -y --releasever=latest --security
+
+RUN \
+    # For extracting software
+    dnf install -y unzip \
+    # Install gnugp2 to perform signature validation
+    && dnf install -y --allowerasing gnupg2 \
+    && dnf clean all
+
+# Curl scanner binary
+RUN curl --retry 12 --retry-all-errors ${SCANNER_ZIP_URL} -o ${SCANNER_ZIP}; \
+    curl --retry 12 --retry-all-errors ${SCANNER_ZIP_URL}.asc -o ${SCANNER_ZIP}.asc
+
+RUN gpg --batch --keyserver "hkps://keys.openpgp.org" --recv-keys "679F1EE92B19609DE816FDE81DB198F93525EC1A"; \
+    gpg --verify ${SCANNER_ZIP}.asc
+
+RUN unzip ${SCANNER_ZIP}; \
+    mv sonar-scanner-${SONAR_SCANNER_VERSION}-linux-x64 ${SONAR_SCANNER_HOME}; \
+    rm ${SCANNER_ZIP}
+
+
+FROM amazonlinux:2023 AS scanner-cli-base
+
+ARG SONAR_SCANNER_HOME=/opt/sonar-scanner
+ENV HOME=/tmp \
     XDG_CONFIG_HOME=/tmp \
     SONAR_SCANNER_HOME=${SONAR_SCANNER_HOME} \
     SONAR_USER_HOME=${SONAR_SCANNER_HOME}/.sonar \
     PATH=${SONAR_SCANNER_HOME}/bin:${PATH} \
     SRC_PATH=/usr/src \
-    SCANNER_WORKDIR_PATH=/tmp/.scannerwork \
-    LANG=en_US.UTF-8 \
-    LC_ALL=en_US.UTF-8
+    SCANNER_WORKDIR_PATH=/tmp/.scannerwork
 
-WORKDIR /opt
+# Copy Scanner installation from builder image
+COPY --from=builder /opt/sonar-scanner /opt/sonar-scanner
+
+RUN \
+    dnf install -y git \
+    && dnf clean all
 
 RUN set -eux; \
-    addgroup -S -g 1000 scanner-cli; \
-    adduser -S -D -u 1000 -G scanner-cli scanner-cli; \
-    apk add --no-cache --virtual build-dependencies wget unzip gnupg; \
-    apk add --no-cache git bash shellcheck "nodejs>=18" openjdk17-jre musl-locales musl-locales-lang tar; \
-    wget -U "scannercli" -q -O /opt/sonar-scanner-cli.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${SONAR_SCANNER_VERSION}.zip; \
-    wget -U "scannercli" -q -O /opt/sonar-scanner-cli.zip.asc https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${SONAR_SCANNER_VERSION}.zip.asc; \
-    for server in $(shuf -e hkps://keys.openpgp.org \
-                            hkps://keyserver.ubuntu.com) ; do \
-        gpg --batch --keyserver "${server}" --recv-keys 679F1EE92B19609DE816FDE81DB198F93525EC1A && break || : ; \
-    done; \
-    gpg --verify /opt/sonar-scanner-cli.zip.asc /opt/sonar-scanner-cli.zip; \
-    unzip sonar-scanner-cli.zip; \
-    rm sonar-scanner-cli.zip sonar-scanner-cli.zip.asc; \
-    mv sonar-scanner-${SONAR_SCANNER_VERSION} ${SONAR_SCANNER_HOME}; \
-    apk del --purge build-dependencies; \
+    groupadd --system --gid 900 scanner-cli; \
```

Comment:
Was it mandatory to change the user and group id? This might again create some issues.